### PR TITLE
libavif: conan v2 support

### PR DESCRIPTION
**Pull request by SpaceIm**
closes https://github.com/conan-io/conan-center-index/issues/13001

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libavif/all/conanfile.py [hunk 1]
```diff
@@ -35,52 +37,71 @@ def config_options(self):
...
+        cmake_layout(self, src_folder="src")
 
     @property
     def _has_dav1d(self):
         return self.options.with_decoder == "dav1d"
 
     def requirements(self):
-        self.requires("libaom-av1/3.1.2")
+        self.requires("libaom-av1/3.4.0")
         self.requires("libyuv/cci.20201106")
```
Comment (SpaceIm) [inline recipes/libavif/all/conanfile.py:58]: By the way, cci versioning of `libyuv` shouldn't have been accepted in https://github.com/conan-io/conan-center-index/pull/3488. There are libyuv versions (there is no tag, but version is tracked upstream: https://chromium.googlesource.com/libyuv/libyuv/+/f71c83552d373f0ff41833b17e2880632d8561d7/include/libyuv/version.h).

**Comment by SpaceIm** [comment]
waiting https://github.com/conan-io/conan-center-index/pull/13010 so that I can change libyuv version to an official one.

File: recipes/libavif/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/libavif/all/conanfile.py:58]: see https://github.com/conan-io/conan-center-index/pull/13010

**Review by uilianries — APPROVED** [review]
LGTM

**Review by toge — APPROVED** [review]

**Pull request merged**